<HTML>
<HEAD>
<meta charset="UTF-8">
<title>FlowException - application</title>
<link rel="stylesheet" href="../../../style.css">
</HEAD>
<BODY>
<a href="../../index.html">application</a>&nbsp;/&nbsp;<a href="../index.html">net.corda.v5.application.flows</a>&nbsp;/&nbsp;<a href="./index.html">FlowException</a><br/>
<br/>
<h1>FlowException</h1>
<code><span class="keyword">open</span> <span class="keyword">class </span><span class="identifier">FlowException</span>&nbsp;<span class="symbol">:</span>&nbsp;<span class="identifier">CordaRuntimeException</span><span class="symbol">, </span><a href="../-identifiable-exception/index.html"><span class="identifier">IdentifiableException</span></a></code>
<p>Exception which can be thrown by a <a href="#">FlowLogic</a> at any point in its logic to unexpectedly bring it to a permanent end.
The exception will propagate to all counterparty flows and will be thrown on their end the next time they wait on a
<a href="../-flow-session/receive.html">FlowSession.receive</a> or <a href="../-flow-session/send-and-receive.html">FlowSession.sendAndReceive</a>. Any flow which no longer needs to do a receive, or has already
ended, will not receive the exception (if this is required then have them wait for a confirmation message).</p>
<p>If the <em>rethrown</em> <a href="./index.html">FlowException</a> is uncaught in counterparty flows and propagation triggers then the exception is
downgraded to an <a href="../-unexpected-flow-end-exception/index.html">UnexpectedFlowEndException</a>. This means only immediate counterparty flows will receive information
about what the exception was.</p>
<p><a href="./index.html">FlowException</a> (or a subclass) can be a valid expected response from a flow, particularly ones which act as a service.
It is recommended a <a href="#">FlowLogic</a> document the <a href="./index.html">FlowException</a> types it can throw.</p>
<h3>Constructors</h3>
<table>
<tbody>
<tr>
<td>
<h4><a href="-init-.html">&lt;init&gt;</a></h4>
</td>
<td>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.v5.application.flows.FlowException$<init>(kotlin.String, kotlin.Throwable)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">, </span><span class="identifier" id="net.corda.v5.application.flows.FlowException$<init>(kotlin.String, kotlin.Throwable)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">)</span></code><br/>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.v5.application.flows.FlowException$<init>(kotlin.String)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">)</span></code><br/>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.v5.application.flows.FlowException$<init>(kotlin.Throwable)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">)</span></code><br/>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="symbol">)</span></code>
<p>Exception which can be thrown by a <a href="#">FlowLogic</a> at any point in its logic to unexpectedly bring it to a permanent end.
The exception will propagate to all counterparty flows and will be thrown on their end the next time they wait on a
<a href="../-flow-session/receive.html">FlowSession.receive</a> or <a href="../-flow-session/send-and-receive.html">FlowSession.sendAndReceive</a>. Any flow which no longer needs to do a receive, or has already
ended, will not receive the exception (if this is required then have them wait for a confirmation message).</p>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.v5.application.flows.FlowException$<init>(kotlin.String, kotlin.Throwable, kotlin.Long)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">, </span><span class="identifier" id="net.corda.v5.application.flows.FlowException$<init>(kotlin.String, kotlin.Throwable, kotlin.Long)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">, </span><span class="identifier" id="net.corda.v5.application.flows.FlowException$<init>(kotlin.String, kotlin.Throwable, kotlin.Long)/originalErrorId">originalErrorId</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-long/index.html"><span class="identifier">Long</span></a><span class="symbol">?</span><span class="symbol">)</span></code></td>
</tr>
</tbody>
</table>
<h3>Properties</h3>
<table>
<tbody>
<tr>
<td>
<h4><a href="error-id.html">errorId</a></h4>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">val </span><span class="identifier">errorId</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-long/index.html"><span class="identifier">Long</span></a><span class="symbol">?</span></code></td>
</tr>
<tr>
<td>
<h4><a href="original-error-id.html">originalErrorId</a></h4>
</td>
<td>
<p>the ID backing <a href="#">getErrorId</a>. If null it will be set dynamically by the flow framework when
    the exception is handled. This ID is propagated to counterparty flows, even when the <a href="./index.html">FlowException</a> is
    downgraded to an <a href="../-unexpected-flow-end-exception/index.html">UnexpectedFlowEndException</a>. This is so the error conditions may be correlated later on.</p>
<code><span class="keyword">var </span><span class="identifier">originalErrorId</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-long/index.html"><span class="identifier">Long</span></a><span class="symbol">?</span></code></td>
</tr>
</tbody>
</table>
</BODY>
</HTML>
